### DISCUSSION TITLE
'Packages which use Internet resources should fail gracefully'

### DISCUSSION THREAD

**Discussion by hturner** [r-devel/cran-cookbook#65] [Community Contributions]
Following a package maintainer being sent an email about this point from the CRAN team, there was a discussion how to fix it on the rOpenSci Slack. This lead to James Balamuta (@coatless ) writing up solutions to this issue in a [blog post](https://blog.thecoatlessprofessor.com/programming/r/api-packages-and-cran-requirements/). I think this is a good candidate for the cookbook and we could adapt James's post (with credit).

A couple of comments:

 - I like the idea of including the revision number of the CRAN policy in the quote
 - Some of his solutions apply specifically to packages accessing Web APIs, but the problem applies to any package that tries to access the internet during a CRAN check, e.g. downloading a csv using a URL.

**Comment by jasdumas** [comment]
Thank you for adding this in Heather!
